### PR TITLE
Simplify default currency fetching for easier code understanding

### PR DESCRIPTION
**Pull request by FabienPapet**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      |Simplify default currency fetching for easier code understanding
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | This PR is basically replacing bad copy pasting by adding a unique method to handle getting default currency. QA can be done by a dev. But you can check some pages where you use the default currency to see if this is still working.
| Possible impacts? | Should have no impact.

### CODE REVIEW

**Comment by zuk3975** [comment]
smth wrong with behats

File: src/Adapter/Cart/CommandHandler/CreateEmptyCustomerCartHandler.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ private function createEmptyCustomerCart(Customer $customer): Cart
 
         $cart->id_shop = $customer->id_shop;
         $cart->id_lang = (int) Configuration::get('PS_LANG_DEFAULT');
-        $cart->id_currency = (int) Configuration::get('PS_CURRENCY_DEFAULT');
+        $cart->id_currency = Currency::getDefaultCurrencyId();
```
Comment (davidglezz) [inline src/Adapter/Cart/CommandHandler/CreateEmptyCustomerCartHandler.php:79]: `use Currency;` should be added at the top.

Comment (zuk3975) [reply src/Adapter/Cart/CommandHandler/CreateEmptyCustomerCartHandler.php:79]: probably its the reason behats fails

**Review by mflasquin — APPROVED** [review]
LGTM

**Review by zuk3975 — APPROVED** [review]

**Review by 0x346e3730 — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @FabienPapet

**Comment by hibatallahAouadni** [comment]
Hello @FabienPapet 

Could you please provide simple steps to test this PR :pray: 
Or do you think this PR should be tested by a Dev?

Thanks!

**Comment by FabienPapet** [comment]
hi @hibatallahAouadni 

This PR is basically replacing bad copy pasting by adding a unique method to handle getting default currency. QA can be done by a dev. But you  *can* check some pages where you use the default currency to see if this is still working.

**Review by davidglezz — APPROVED** [review]

**Comment by paulnoelcholot** [comment]
Hello @FabienPapet,

I have tested this PR and it's OK for me, I have found no regression :tada: 

Thanks you !

**Review by paulnoelcholot — APPROVED** [review]

**Comment by zuk3975** [comment]
Thanks @FabienPapet :tada:

**Pull request merged**